### PR TITLE
verify maxPartID in object options helpers

### PR DESCRIPTION
**Pull request by jiuker**
## Description

```go
	partIDString := r.Form.Get(xhttp.PartNumber)

	partID, err := strconv.Atoi(partIDString)
	if err != nil {
		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrInvalidPart), r.URL)
		return
	}

	// check partID with maximum part ID for multipart objects
	if isMaxPartID(partID) {
		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrInvalidMaxParts), r.URL)
		return
	}
```
We can do like this?
Close it if on purpose.
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
In getOpts() the part number is validated at lower layer, so what you are adding is fine.

**Comment by jiuker** [comment]
@harshavardhana close this if on purpose

File: cmd/object-api-options.go [hunk 1]
```diff
@@ -89,6 +89,9 @@ func getOpts(ctx context.Context, r *http.Request, bucket, object string) (Objec
 		if err != nil {
 			return opts, err
 		}
+		if isMaxPartID(partNumber) {
+			return opts, fmt.Errorf("ErrInvalidMaxParts")
```
Comment (harshavardhana) [inline cmd/object-api-options.go]: We need to return correct errors here @jiuker

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/typed-errors.go [hunk 2]
```diff
@@ -112,3 +112,6 @@ var errIAMNotInitialized = errors.New("IAM sub-system is being initialized, plea
 
 // error returned when upload id not found
 var errUploadIDNotFound = errors.New("Specified Upload ID is not found")
+
+// error returned when request.From http.PartNumber is greater than the maximum allowed ID.
```
Comment (harshavardhana) [inline cmd/typed-errors.go]:
```suggestion
// error returned when PartNumber is greater than the maximum allowed 10000 parts
```

File: cmd/typed-errors.go [hunk 3]
```diff
@@ -112,3 +112,6 @@ var errIAMNotInitialized = errors.New("IAM sub-system is being initialized, plea
...
+var errInvalidMaxParts = errors.New("Part ID is greater than the maximum allowed ID.")
```
Comment (harshavardhana) [inline cmd/typed-errors.go]:
```suggestion
var errInvalidMaxParts = errors.New("Part number is greater than the maximum allowed 10000 parts")
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]

**Pull request merged**